### PR TITLE
[SPARK-47686][SQL][TESTS] Use `=!=` instead of `!==` in `JoinHintSuite`

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr use `=!=` instead of `!==` in `JoinHintSuite`. `!==`  is a deprecated API since 2.0.0, and its test already exists in `DeprecatedAPISuite`.

### Why are the changes needed?
Clean up deprecated API  usage.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass GitHub Actions

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/JoinHintSuite.scala [hunk 1]
```diff
@@ -695,11 +695,11 @@ class JoinHintSuite extends PlanTest with SharedSparkSession with AdaptiveSparkP
     val hintAppender = new LogAppender(s"join hint check for equi-join")
     withLogAppender(hintAppender, level = Some(Level.WARN)) {
       assertBroadcastNLJoin(
-        df1.hint("SHUFFLE_HASH").join(df2, $"a1" !== $"b1"), BuildRight)
+        df1.hint("SHUFFLE_HASH").join(df2, $"a1"=!= $"b1"), BuildRight)
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/JoinHintSuite.scala:698]: Space? `"=!=` -> `" =!=`

File: sql/core/src/test/scala/org/apache/spark/sql/JoinHintSuite.scala [hunk 2]
```diff
@@ -695,11 +695,11 @@ class JoinHintSuite extends PlanTest with SharedSparkSession with AdaptiveSparkP
...
     }
     withLogAppender(hintAppender, level = Some(Level.WARN)) {
       assertBroadcastNLJoin(
-        df1.join(df2.hint("MERGE"), $"a1" !== $"b1"), BuildRight)
+        df1.join(df2.hint("MERGE"), $"a1" =!= $"b1"), BuildRight)
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/JoinHintSuite.scala:702]: Is this the last instance in Spark repo?

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Only two minor comments.

File: sql/core/src/test/scala/org/apache/spark/sql/JoinHintSuite.scala (hunk 1 above)
Comment (LuciferYang) [reply sql/core/src/test/scala/org/apache/spark/sql/JoinHintSuite.scala]: done

File: sql/core/src/test/scala/org/apache/spark/sql/JoinHintSuite.scala (hunk 2 above)
Comment (LuciferYang) [reply sql/core/src/test/scala/org/apache/spark/sql/JoinHintSuite.scala:702]: image

Yes, other calls are in the DeprecatedAPISuite, we should keep them.

**Comment by dongjoon-hyun** [comment]
Merged to master.

**Pull request closed without merging**